src/vsr/replica_test.zig

```diff
@@ -1506,29 +1507,18 @@ const TestReplicas = struct {
         return paths;
     }
 
-    fn expect_equal_grid(want: TestReplicas, got: TestReplicas) !void {
-        assert(want.replicas.count() == 1);
-        assert(got.replicas.count() > 0);
-
-        const want_replica: *const Cluster.Replica = &want.cluster.replicas[want.replicas.get(0)];
-
-        for (got.replicas.const_slice()) |replica_index| {
-            const got_replica: *const Cluster.Replica = &got.cluster.replicas[replica_index];
+    fn expect_sync_done(t: TestReplicas) !void {
+        assert(t.replicas.count() > 0);
 
-            var address: u64 = 1;
-            while (address <= vsr.superblock.grid_blocks_max) : (address += 1) {
-                const address_free = want_replica.superblock.free_set.is_free(address);
-                assert(address_free == got_replica.superblock.free_set.is_free(address));
-                if (address_free) continue;
+        for (t.replicas.const_slice()) |replica_index| {
+            const replica: *const Cluster.Replica = &t.cluster.replicas[replica_index];
+            assert(replica.sync_content_done());
 
-                const block_want = want_replica.superblock.storage.grid_block(address).?;
-                const block_got = got_replica.superblock.storage.grid_block(address).?;
+            // If the replica has finished syncing, but not yet checkpointed, then it might not have
+            // updated its sync_op_max.
+            maybe(replica.superblock.staging.vsr_state.sync_op_max > 0);
 
-                try expectEqual(
-                    std.mem.bytesToValue(vsr.Header, block_want[0..@sizeOf(vsr.Header)]),
-                    std.mem.bytesToValue(vsr.Header, block_got[0..@sizeOf(vsr.Header)]),
-                );
-            }
+            try t.cluster.storage_checker.replica_sync(&replica.superblock);
         }
```

Comment:
The previous strategy for checking the grid doesn't always work (we were "lucky" before) for the same reason that the storage checker cannot verify determinism except at checkpoints:
Manifest block addresses are acquired and then held onto for potentially multiple bars before they are written.

Instead, we just trigger the storage checker, which already knows how to verify all of this properly!